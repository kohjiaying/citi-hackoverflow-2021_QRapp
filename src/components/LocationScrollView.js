import React, { Component } from 'react';
import { Text, Image, View, StyleSheet, ScrollView, Dimensions, TouchableOpacity, Modal, Alert, Pressable } from 'react-native';

class LocationScrollView extends Component {
   
   state = {
		modalVisible: false,
      names: [
         {'name': 'Tampines', 'id': 1},
         {'name': 'Simei', 'id': 2},
         {'name': 'asdwfd', 'id': 3},
         {'name': 'Mary', 'id': 4},
         {'name': 'Daniel', 'id': 5}
      ],
	  
	  selectedItem: {'name': 'Tampines', 'id': 1},
	   tempStr: 'dummy1'
   }
   
   setModalVisible(visible) {
    	this.setState({modalVisible: visible});
  }
  setSelectedItem(newItem) {
    	this.setState({selectedItem: newItem});
  }
   
   
   render() {
      return (
         <View style={styles.mainContainer}>
		    <Text style = {styles.header}> Stores Around You </Text>
            <ScrollView horizontal={true}>
				{
                  this.state.names.map((item, index) => (
                     <TouchableOpacity key = {item.id} 
							onPress={() => {
                    			this.setModalVisible(true);
			              		this.setSelectedItem(item); 
                  }}>
                    		<Image 
								style = {styles.Logo}
								source = {{uri: 'https://reactnative.dev/img/tiny_logo.png'}}/>
							<Text>{item.name}</Text>
							
                		</TouchableOpacity>
                  ))
                }
            </ScrollView>
			<Modal
			          	animationType="slide"
			          	transparent={true}
			          	visible={this.state.modalVisible}
			          	onRequestClose={() => {
					    	this.setModalVisible(!this.state.modalVisible);
					      }}>
			          	<View style={styles.modalcontainer}>
						<View style={styles.modalcard}>
						<Text style={{fontSize: 20, padding: 5}}>{this.state.selectedItem.name}</Text>
                		<Text style={styles.space}></Text>
                  		<Text style={styles.space}></Text>
						<Pressable
							style={[styles.button, styles.buttonClose]}
							onPress={() => {
								this.setModalVisible(!this.state.modalVisible);
							}}
						>
							<Text style={styles.textStyle}>Close</Text>
						</Pressable>
						</View>
			          	</View>
					</Modal>
         </View>
      )
   }
}
export default LocationScrollView

const styles = StyleSheet.create ({
	scrollView : {}, 
	mainContainer : {
		marginTop: 10
	}, 
	header: {
		fontSize: 20
	},
    item: {
      padding: 75,
      margin: 2,
      borderColor: '#414757',
      borderWidth: 1,
      backgroundColor: '#560CCE',
   },
   logo: {
    width: 66,
    height: 58,
	},
	modalcontainer: {
		alignItems: 'center',
        justifyContent: 'center',
		flex: 1 
	},
	modalcard: {
		margin: 5,
		backgroundColor: "white",
		borderRadius: 0,
		padding: 10,
		width: 350,
		height: 500,
		shadowColor: "#000",
		shadowOffset: {
		width: 100,
		height: 100
		},
		shadowOpacity: 0.25,
		shadowRadius: 4,
		elevation: 5,
		alignItems: 'center',
	},
	button: {
    alignItems: 'center',
    backgroundColor: '#560CCE',
    padding: 10,
	  width: '100%'
    },
	buttonClose: {
	  backgroundColor: "#560CCE",
    },
  buttonText: {
    textAlign: 'center',
    padding: 20,
    color: 'white'
  },
  space: {
    width: 20,
    height: 20,
  },
  logo: {
    width: 100,
    height: 100,
  }
})