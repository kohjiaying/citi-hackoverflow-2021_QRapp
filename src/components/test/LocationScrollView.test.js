import { JestEnvironment } from '@jest/environment';
import React from 'react';
import renderer from 'react-test-renderer';
import LocationScrollView from '../LocationScrollView';


describe('<MyLocationScrollView />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<LocationScrollView />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });