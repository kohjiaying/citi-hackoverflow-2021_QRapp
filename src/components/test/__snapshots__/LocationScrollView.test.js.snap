// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyLocationScrollView /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
     Stores Around You 
  </Text>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Bedok
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Changi
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Simei
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Tampines
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 0,
            "elevation": 5,
            "height": 450,
            "margin": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 100,
              "width": 100,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 0,
              "padding": 5,
              "textAlign": "center",
            }
          }
        />
        <RCTScrollView>
          <View>
            <View
              onPress={[Function]}
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  style={
                    Object {
                      "borderColor": "#414757",
                      "borderWidth": 1,
                      "height": 150,
                      "margin": 2,
                      "width": 300,
                    }
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  style={
                    Object {
                      "borderColor": "#414757",
                      "borderWidth": 1,
                      "height": 150,
                      "margin": 2,
                      "width": 300,
                    }
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  style={
                    Object {
                      "borderColor": "#414757",
                      "borderWidth": 1,
                      "height": 150,
                      "margin": 2,
                      "width": 300,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#560CCE",
              "borderRadius": 5,
              "borderWidth": 2,
              "margin": 5,
              "padding": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#560CCE",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`<MyVoucherScrollView /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
     Stores Around You 
  </Text>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Bedok
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Changi
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Simei
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#560CCE",
            "borderColor": "#414757",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 3,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderColor": "#414757",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 150,
              "width": 150,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Tampines
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 0,
            "elevation": 5,
            "height": 450,
            "margin": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 100,
              "width": 100,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginTop": 0,
              "padding": 5,
              "textAlign": "center",
            }
          }
        />
        <RCTScrollView>
          <View>
            <View
              onPress={[Function]}
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  style={
                    Object {
                      "borderColor": "#414757",
                      "borderWidth": 1,
                      "height": 150,
                      "margin": 2,
                      "width": 300,
                    }
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  style={
                    Object {
                      "borderColor": "#414757",
                      "borderWidth": 1,
                      "height": 150,
                      "margin": 2,
                      "width": 300,
                    }
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  style={
                    Object {
                      "borderColor": "#414757",
                      "borderWidth": 1,
                      "height": 150,
                      "margin": 2,
                      "width": 300,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#560CCE",
              "borderRadius": 5,
              "borderWidth": 2,
              "margin": 5,
              "padding": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#560CCE",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
